### PR TITLE
refactor(gitlab): Use new http wrapper for Gitlab platform

### PR DESCRIPTION
**Pull request by zharinov**
Ref: #5865

### CODE REVIEW

**Comment by zharinov** [comment]
Do we have good example repos to test against?

**Comment by zharinov** [comment]
Probably, I can just push my existing test repo to new Gitlab remote

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by zharinov** [comment]
Seems to be working with test repo: https://gitlab.com/zharinov/renovate-test/-/merge_requests

File: lib/platform/gitlab/index.ts [hunk 1]
```diff
@@ -959,29 +1025,29 @@ export async function ensureCommentRemoval({
...
-    const res = await api.get(urlString, { paginate: true });
-    return res.body.map(mapPullRequests);
+    const res = await gitlabApi.getJson<
+      {
+        iid: number;
+        source_branch: string;
+        title: string;
+        state: string;
+        created_at: string;
+      }[]
```
Comment (viceice) [inline lib/platform/gitlab/index.ts:1042]: We sould extract a type for this

File: lib/platform/gitlab/index.ts [hunk 2]
```diff
@@ -686,7 +731,7 @@ export async function setBranchStatus({
 
 export async function getIssueList(): Promise<any[]> {
   if (!config.issueList) {
-    const res = await api.get(
+    const res = await gitlabApi.getJson<{ iid: number; title: string }[]>(
```
Comment (viceice) [inline lib/platform/gitlab/index.ts:734]: Create a type for this. maybe we should move all gitlab api types to `./types.ts`

File: lib/platform/gitlab/index.ts [hunk 3]
```diff
@@ -430,7 +451,18 @@ export async function createPr({
 export async function getPr(iid: number): Promise<Pr> {
   logger.debug(`getPr(${iid})`);
   const url = `projects/${config.repository}/merge_requests/${iid}?include_diverged_commits_count=1`;
-  const pr = (await api.get(url)).body;
+  const pr = (
+    await gitlabApi.getJson<
+      Pr & {
```
Comment (viceice) [inline lib/platform/gitlab/index.ts:456]: We should have a helper function for mapping Pr to have proper types, see comments below

**Review by viceice — COMMENTED** [review]
Some suggesttions. Maybe @zharinov can do the type optimizations in a later pr?

**Comment by zharinov** [comment]
> Some suggesttions. Maybe @zharinov can do the type optimizations in a later pr?

@viceice Sure, and for other platforms as well

**Comment by zharinov** [comment]
Sample repo: https://gitlab.com/zharinov/renovate-test/-/merge_requests/

**Pull request merged**